presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -84,15 +86,17 @@
     private final AtomicLong finishUserNanos = new AtomicLong();
 
     private final AtomicLong memoryReservation = new AtomicLong();
-    private final AtomicLong revocableMemoryReservation = new AtomicLong();
+    @GuardedBy("this")
```

Comment:
Well, generally yes, but isn't that what `@GuardedBy` is for? I don't think splitting this class is a good idea. I can make it thoroughly synchronized, but I am not convinced. @losipiuk what is your opinion?